test/pg-cdc/pg-cdc.td

```diff
@@ -0,0 +1,392 @@
+Source error: File IO: error connecting to server: Connection refused (os error 111)
+
+> CREATE MATERIALIZED SOURCE "no_such_user"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=no_such_user password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_user;
+Source error: File IO: db error: FATAL: role "no_such_user" does not exist
+
+> CREATE MATERIALIZED SOURCE "no_such_password"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=no_such_password dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_password;
+internal error: column 0 is not of expected type ColumnType { nullable: true, scalar_type: Timestamp }: Int32(1)
+
+> CREATE MATERIALIZED SOURCE "no_such_dbname"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=no_such_dbname'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_dbname;
+Source error: File IO: db error: FATAL: database "no_such_dbname" does not exist
+
+> CREATE MATERIALIZED SOURCE "no_such_publication"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'no_such_publication' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_publication;
+internal error: column 0 is not of expected type ColumnType { nullable: true, scalar_type: Timestamp }: Int32(1)
+
+> CREATE MATERIALIZED SOURCE "no_such_namespace"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'no_such_namespace' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_namespace;
+Source error: File IO: table not found in the upstream catalog
+
+> CREATE MATERIALIZED SOURCE "no_such_table"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'no_such_table' (pk INTEGER, f2 TEXT);
+
+! SELECT * FROM no_such_table;
+Source error: File IO: table not found in the upstream catalog
+
+> CREATE MATERIALIZED SOURCE "no_such_column"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (no_such_column INTEGER, f2 TEXT);
+
+# ! SELECT * FROM no_such_column;
+# error expected here but success is returned
+
+> CREATE MATERIALIZED SOURCE "column_type_mismatch"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM column_type_mismatch;
+internal error: column 0 is not of expected type ColumnType { nullable: true, scalar_type: Timestamp }: Int32(1)
```

Comment:
This error could have been better even though frankly speaking it does contain some information as to what is going on.